internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,190 @@
+package accesslist
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cli/user"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/10gen/realm-cli/internal/utils/flags"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+const (
+	headerDeleted = "Deleted"
+	headerDetails = "Details"
+)
+
+var (
+	deleteTableHeaders = []string{headerAddress, headerComment, headerDeleted, headerDetails}
+)
+
+type deleteAllowedIPOutputs []deleteAllowedIPOutput
+
+type deleteAllowedIPOutput struct {
+	allowedIP realm.AllowedIP
+	err       error
+}
+type deleteInputs struct {
+	cli.ProjectInputs
+	Addresses []string
+}
+
+// CommandMetaDelete is the command meta for the `accesslist delete` command
+var CommandMetaDelete = cli.CommandMeta{
+	Use:         "delete",
+	Display:     "accesslist delete",
+	Description: "Delete an IP address or CIDR block from the Access List of your Realm app",
+	HelpText: `Removes an existing entry from the Access List of your Realm app. You will be
+prompted to select an IP address or CIDR block if none are provided in the
+initial command.`,
+}
+
+// CommandDelete is the `accesslist delete` command
+type CommandDelete struct {
+	inputs deleteInputs
+}
+
+// Flags is the command flags
+func (cmd *CommandDelete) Flags() []flags.Flag {
+	return []flags.Flag{
+		cli.AppFlagWithContext(&cmd.inputs.App, "to remove an entry in its Access List"),
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address(es) or CIDR block(s) to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selectedAllowedIPs, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selectedAllowedIPs) == 0 {
+		return errors.New("No IP addresses or CIDR blocks to delete")
```

Comment:
generally keep capitalization out of error messages in go (unless its a proper noun)
```suggestion
		return errors.New("no IP addresses or CIDR blocks to delete")
```